google-home-community.groovy

```diff
@@ -4228,16 +4262,42 @@ private traitFromSettings_SensorState(traitName) {
     ]
 
     settings."${traitName}.sensorTypes"?.each { sensorType ->
+        // default to true
         def sensorMapping = [
-            descriptiveState:               [:],
-            descriptiveAttribute:           [:],
-            numericAttribute:               [:],
-            numericUnits:                   [:],
+             reportsDescriptiveState:            true,
+             reportsNumericState:                true,
         ]
-        sensorMapping.descriptiveState = settings."${traitName}.sensorTypes.${sensorType}.availableStates"
-        sensorMapping.descriptiveAttribute = settings."${traitName}.sensorTypes.${sensorType}.descriptiveAttribute"
-        sensorMapping.numericAttribute = settings."${traitName}.sensorTypes.${sensorType}.numericAttribute"
-        sensorMapping.numericUnits = settings."${traitName}.sensorTypes.${sensorType}.numericUnits"
+
+        // build the descriptive traits
+        if (GOOGLE_SENSOR_STATES[sensorType].descriptiveState) {
+             if (GOOGLE_SENSOR_STATES[sensorType].numericAttribute != "") {
+                  sensorMapping.reportsDescriptiveState = settings."${traitName}.sensorTypes.${sensorType}.reportsDescriptiveState"
```

Comment:
You could set `reportsDescriptiveState` to true by default here and avoid needing to pepper a bunch of `!= false` comparisons around:

```suggestion
                  sensorMapping.reportsDescriptiveState = settings."${traitName}.sensorTypes.${sensorType}.reportsDescriptiveState" ?: true
```

Reply:
Much cleaner.  Done.